File: 251119/최대로 겹치는 구간/maximum-overlapped-segments.py
n = int(input())
segments = [tuple(map(int, input().split())) for _ in range(n)]

arr = [0 for _ in range(200)]
offset = 100
for s, e in segments:
    s += 100
    e += 100
    for i in range(s, e):
        arr[i] += 1

print(max(arr))

File: 260308/최대로 겹치는 구간/maximum-overlapped-segments.js
const MAX_SIZE = 200;
const OFFSET = 100;

const fs = require("fs");
const input = fs.readFileSync(0).toString().trim().split('\n');

const n = Number(input[0]);
const segments = [];
for (let i = 1; i <= n; i++) {
    segments.push(input[i].split(' ').map(Number));
}

let straightLine = Array(MAX_SIZE + 1).fill(0);
for (let [start, end] of segments) {
    start += OFFSET; // 음수 인덱스 방지
    end += OFFSET;
    for (let i = start; i < end; i++) {
        straightLine[i] += 1;
    }
}

console.log(Math.max(...straightLine));